x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -481,13 +481,17 @@ export class KnowledgeBaseService {
   }): Promise<{
     entries: RecalledEntry[];
   }> => {
-    this.dependencies.logger.debug(`Recalling entries from KB for queries: "${queries}"`);
+    const mappedQueries = queries.map((query) =>
+      typeof query === 'string' ? { boost: 1, text: query } : query
```

Comment:
When is `query` a string and not an object?

Reply:
Leftover from merging your work into mine I think. Will fix!